src/Columns/ColumnObject.h

```diff
@@ -145,11 +145,12 @@ class ColumnObject final : public COWHelper<IColumnHelper<ColumnObject>, ColumnO
     void getPermutation(PermutationSortDirection, PermutationSortStability, size_t, int, Permutation &) const override;
     void updatePermutation(PermutationSortDirection, PermutationSortStability, size_t, int, Permutation &, EqualRanges &) const override {}
 
-    /// Values of ColumnObject are not comparable.
+    /// Values of ColumnObject are not comparable for less and greater functions.
```

Comment:
Do we force it somewhere, so other operators shows proper error?

Reply:
Unfortunately no, for not comparable types like JSON/Object/AggregateFunction/SimpleAggregateFunction we just fallback to generic implementation of `less/greater/etc` functions that uses `IColumn::compareAt` (and for these types it always returns 0 except JSON now that returns 0 for equal and 1 for all other cases).

I would change it and throw an exception in functions `less/greater/etc` if type is not comparable. And maybe in ORDER BY as well. But it will be backward incompatible change, so let me do it in a separate PR under a setting.